opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -658,53 +651,24 @@ def get_tracer(
     This function is a convenience wrapper for
     opentelemetry.trace.tracer_provider().get_tracer
     """
-    return tracer_provider().get_tracer(
+    return get_tracer_provider().get_tracer(
         instrumenting_module_name, instrumenting_library_version
     )
 
 
-def tracer_provider() -> TracerProvider:
-    """Gets the current global :class:`~.TracerProvider` object.
-
-    If there isn't one set yet, a default will be loaded.
-    """
-    global _TRACER_PROVIDER, _TRACER_PROVIDER_FACTORY  # pylint:disable=global-statement
-
-    if _TRACER_PROVIDER is None:
-        # pylint:disable=protected-access
-        try:
-            _TRACER_PROVIDER = loader._load_impl(
-                TracerProvider, _TRACER_PROVIDER_FACTORY  # type: ignore
-            )
-        except TypeError:
-            # if we raised an exception trying to instantiate an
-            # abstract class, default to no-op tracer impl
-            logger.warning(
-                "Unable to instantiate TracerProvider from factory.",
-                exc_info=True,
-            )
-            _TRACER_PROVIDER = DefaultTracerProvider()
-        del _TRACER_PROVIDER_FACTORY
-
-    return _TRACER_PROVIDER
-
+def set_tracer_provider(tracer_provider: TracerProvider) -> None:
+    """Sets the current global :class:`~.TracerProvider` object."""
+    global _TRACER_PROVIDER  # pylint: disable=global-statement
+    _TRACER_PROVIDER = tracer_provider
 
-def set_preferred_tracer_provider_implementation(
-    factory: ImplementationFactory,
-) -> None:
-    """Set the factory to be used to create the global TracerProvider.
 
-    See :mod:`opentelemetry.util.loader` for details.
+def get_tracer_provider() -> TracerProvider:
+    """Gets the current global :class:`~.TracerProvider` object."""
+    global _TRACER_PROVIDER  # pylint: disable=global-statement
 
-    This function may not be called after a tracer is already loaded.
-
-    Args:
-        factory: Callback that should create a new :class:`TracerProvider`
-            instance.
-    """
-    global _TRACER_PROVIDER_FACTORY  # pylint:disable=global-statement
-
-    if _TRACER_PROVIDER:
-        raise RuntimeError("TracerProvider already loaded.")
+    if _TRACER_PROVIDER is None:
+        _TRACER_PROVIDER = (
+            Configuration().tracer_provider  # type: ignore # pylint: disable=no-member
+        )
 
-    _TRACER_PROVIDER_FACTORY = factory
+    return _TRACER_PROVIDER  # type: ignore
```

Comment:
The behaviour of the code before was to prevent someone from setting a provider multiple times. I don't know enough about the context of that code to understand whether or not that behaviour is still desirable, but it looks like the behaviour.